src/diff.rs

```diff
@@ -152,10 +154,19 @@ impl<'a> DiffState<'a> {
     }
 }
 
-pub fn print_diff(expected: &str, actual: &str) {
-    let mut state = DiffState::default();
-    for row in lines(expected, actual) {
-        state.row(row);
+pub fn print_diff(expected: &[u8], actual: &[u8]) {
+    if let (Ok(expected), Ok(actual)) = (std::str::from_utf8(expected), std::str::from_utf8(actual))
+    {
+        let mut state = DiffState::default();
+        for row in lines(expected, actual) {
+            state.row(row);
+        }
+        state.finish();
+    } else {
+        eprint!("{}", "-".red());
+        std::io::stderr().write(expected).unwrap();
```

Comment:
This will print many lines, right? That would look rather strange I think.
```
-the_output
some non-utf-8
+the_output
other non-utf-8
```
Maybe it should just do the diff on the utf-8-lossy result but print a warning saying the diff might miss some information?